modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/index/hash/io/HashIndexTreeIo.java

```diff
@@ -43,25 +52,27 @@
  * <p>Defines a following data layout:
  * <ul>
  *     <li>Index columns hash - int (4 bytes);</li>
- *     <li>Index columns link - long (6 bytes);</li>
+ *     <li>Inlined index columns size - short (2 bytes), no more than the {@link InlineUtils#MAX_BINARY_TUPLE_INLINE_SIZE}, the most
+ *     significant (sign) bit is used to determine whether there is a link to the index columns, if negative then no link;</li>
```

Comment:
Please remove reference to most significant bit and only leave the information about sign to avoid confusion. `-x != (x | ((short) 0x80)`, so it's not only the sign bit, but other bits are changed as well

Reply:
Fix it